src/org/kie/jenkins/MavenCommand.groovy

```diff
@@ -106,17 +145,36 @@ def class MavenCommand {
             .withOptions(this.mavenOptions)
             .withPropertyMap(this.properties)
             .withProfiles(this.profiles)
+            .withDependencyRepositoriesInSettings(this.dependenciesRepositories)
+        if(this.settingsXmlConfigFileId){
+            newCmd.withSettingsXmlId(this.settingsXmlConfigFileId)
+        }
         if(this.settingsXmlPath){
             newCmd.withSettingsXmlFile(this.settingsXmlPath)
         }
         if(this.logFileName){
             newCmd.withLogFileName(this.logFileName)
         }
+        if(this.directory){
+            newCmd.inDirectory(this.directory)
+        }
+        if(this.returnStdout){
+            newCmd.returnOutput()
+        }
         return newCmd
     }
 
     MavenCommand returnOutput(){
         this.returnStdout = true
         return this
     }
+
+    private void setRepositoryInSettings(String settingsFilePath, String repoId, String repoUrl) {
+        def depsRepositoryContent = "<id>${repoId}</id><name>${repoId}</name><url>${repoUrl}</url><layout>default</layout><snapshots><enabled>true</enabled></snapshots><releases><enabled>true</enabled></releases>"
+        steps.sh "sed -i 's|<repositories>|<repositories><repository>${depsRepositoryContent}</repository>|g' ${settingsFilePath}"
+        steps.sh "sed -i 's|<pluginRepositories>|<pluginRepositories><pluginRepository>${depsRepositoryContent}</pluginRepository>|g' ${settingsFilePath}"
+        steps.sh "sed -i 's|</mirrorOf>|,!${repoId}</mirrorOf>|g' ${settingsFilePath}"
```

Comment:
What if we replaced this by a single step.sh call which does all the replacements? Thinking about atomicity and how it appears in the Jenkins output logs.

Reply:
I was thinking about "labellizing" some Maven commands to have a better view of what is happening directly in Jenkins.
But indeed this could be done in one call

Reply:
updated